### PR TITLE
new harbor with v2.4.2

### PR DESCRIPTION
**Pull request by beratio**
This PR aims to update registry module with Harbor version 2.4.2

### CODE REVIEW

**Comment by ralgozino** [comment]
Hi @beratio ! Thank you for taking care of this, is the PR ready to be reviewed?

**Comment by ralgozino** [comment]
writing here what @beratio told me: this has been tested with a Fury cluster running kubernetes 1.20 and on Kind with 1.18

File: docs/releases/v1.0.1.md [hunk 1]
```diff
@@ -1,13 +0,0 @@
-# Release notes
```
Comment (nutellinoit) [inline docs/releases/v1.0.1.md:1]: why all the old release notes are deleted?

**Review by nutellinoit — CHANGES_REQUESTED** [review]

File: docs/releases/v1.0.1.md (hunk 1 above)
Comment (ralgozino) [reply docs/releases/v1.0.1.md:1]: Can you please add also the release notes for the changes introduced by this PR? use one of the previous as an example.

I understand there are breaking changes with this update, in that case, the file should be named `v2.0.0.md`, 
If the changes are not breaking we can use `v1.3.0.md` instead.

Thank you!

**Review by ralgozino — CHANGES_REQUESTED** [review]
needs work on the release notes

**Review by ralgozino — APPROVED** [review]
I've pushed some changes to this PR:
- restored the old release notes
- added the release notes for 1.3.0
- ported the main readme to our current standard

**Comment by ralgozino** [comment]
I remembered that the upgrade from harbor 2.2.2 to 2.4.2 has breaking changes in between that we are not handling. Marking this update to the module as 2.0.0 instead of 1.3.0 in consequence.

File: docs/releases/v2.0.0.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+## Changelog
+
+- Update Harbor from version `v2.2.2` to `v2.4.2`.
+
+## Upgrade path
+
+There's no supported upgrade path for this module from `v1.2.0` to `v2.0.0`, there are several breaking changes in Harbor from v2.2.2 to v2.4.2, especially in [Harbor v2.3.0 the Database schema has changed](https://github.com/goharbor/harbor/releases/tag/v2.3.0), please read carefully the release notes from upstream and plan accordingly.
+
+The recommended approach is a lift and shift migration.
```
Comment (nutellinoit) [inline docs/releases/v2.0.0.md:13]: What is the procedure for the lift and shift?

File: README.md [hunk 3]
```diff
@@ -1,55 +1,110 @@
...
-could hit the following [issue](https://github.com/goharbor/harbor/issues/14932).
+4. Define a `kustomization.yaml` that includes the `./vendor/katalog/registry/harbor` directory as resource.
 
+```yaml
+resources:
+- ./vendor/katalog/registry/harbor
+```
 
-## Examples
+5. Apply the necessary patches.
```
Comment (nutellinoit) [inline README.md:72]: Maybe we can link the example here

File: docs/COMPATIBILITY_MATRIX.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+- :warning: Has issues
+- :x: Incompatible
+
+## Warning while upgrading from 1.X to 1.2
+
+If you are using notary in your Harbor setup and you updated the setup from 1.x to 1.2 you could hit the following [issue](https://github.com/goharbor/harbor/issues/14932).
+
+## Warning while upgrading from 1.x to 2.0
+
+There's no supported upgrade path. A lift and shift migration is the recommended approach.
```
Comment (nutellinoit) [inline docs/COMPATIBILITY_MATRIX.md:23]:
```suggestion
There's no supported upgrade path. See [release note v2.0.0](releases/v2.0.0.md) for more informations.
```

File: docs/releases/v2.0.0.md (hunk 2 above)
Comment (ralgozino) [reply docs/releases/v2.0.0.md:13]: pinging @beratio here. Do we have something written that we can reuse?

**Review by nutellinoit — APPROVED** [review]
LGTM 🎉

**Comment by ralgozino** [comment]
I'll merge this PR because I consider it done. The missing guide to migrate from 2.2.2 to 2.4.2 will be tracked in #14 and block the release of the v2.0.0 of the module.

**Pull request merged**